avocado/core/settings.py

```diff
@@ -387,6 +397,16 @@ def add_argparser_to_option(self, namespace, parser, long_arg=None,
             Whether the same option may be available on different parsers.
             This is useful when the same option is available on different
             commands, such as "avocado run" or "avocado list".
+
+        argparse_type :
+            A possibly different type for the command line handling of
+            an option.  For instance, when an option has a "key_type" of "list",
+            its respective configuration file entry will expect a string that
+            can be evaluated to a Python list, but that is far from convenient
+            to set on the command line.  With this argument, a function that
+            will, for instance, split a comma separated list may be used,
+            resulting in command line users being able to provide convenient
+            input.
```

Comment:
I'm missing here, where you are using `self.argparse_type`, probably I'm missing the part that you are calling this method.

Reply:
That is called on `add_argparser_to_option()`, but AFAICT, it needs to set that in the `ConfigOption` itself, because it (`ConfigOption`) uses it in its `_update_argparser()` method.